micro-rdk/src/common/app_client.rs

```diff
@@ -284,6 +286,37 @@ impl AppClient {
         } else {
             None
         };
+
+        #[cfg(feature = "esp32")]
+        {
+            // If the current datetime has not already been set, we use the datetime from
+            // the config response to set the time of day on the device. This may be replaced
+            // by calls to an NTP server in the future
+            let local_dt = Local::now().fixed_offset();
+            // Viam does not pre-exist the year 2020, so if the year is before that
+            // at the very least the current time is wrong and needs to be corrected
+            if local_dt.year() < 2020 {
+                if let Some(current_dt) = datetime {
+                    use esp_idf_svc::sys::{settimeofday, timeval};
+                    let tz = chrono_tz::Tz::UTC;
+                    std::env::set_var("TZ", tz.name());
+                    let tv_sec = current_dt.timestamp() as i32;
+                    let tv_usec = current_dt.timestamp_subsec_micros() as i32;
+                    let current_timeval = timeval { tv_sec, tv_usec };
+                    let res = unsafe {
+                        settimeofday(&current_timeval as *const timeval, std::ptr::null())
+                    };
+                    if res != 0 {
+                        log::error!(
+                            "could not set time of day for timezone {:?} and timestamp {:?}",
+                            tz.name(),
+                            current_dt
+                        );
+                    }
+                }
+            }
```

Comment:
Any thoughts on this? A fine answer might be "we'd want ntp for that". Or we could log a warning if these values differed by a significant amount? How much might that be?

Reply:
I was thinking that we'd hold off doing this kind of thing until we provide ntp support because it can be used to deal with these issues more elegantly